moto/ec2/models/subnets.py

```diff
@@ -317,8 +317,9 @@ def create_subnet(
         ipv6_cidr_block: Optional[str] = None,
         availability_zone: Optional[str] = None,
         availability_zone_id: Optional[str] = None,
-        tags: Optional[List[Dict[str, str]]] = None,
+        tags: Optional[dict[str, dict[str, str]]] = None,
```

Comment:
I think these should be uppercase:

```suggestion
        tags: Optional[Dict[str, Dict[str, str]]] = None,
```

https://github.com/getmoto/moto/actions/runs/6384917966/job/17330933479?pr=6860